### PR TITLE
fix(export): log DistanceExporter metric computation failures instead of swallowing them

### LINKED ISSUE #874: bug(export): DistanceExporter silently swallows metric computation errors, producing None values in exported data that are indistinguishable from legitimate "no path" results
## Summary

`semantica/export/distance_exporter.py` computes pairwise graph metrics (hop count, weighted distance, semantic similarity, betweenness centrality) for export to CSV, JSONL, and DataFrame formats. Four private helper methods each catch all exceptions and return a silent `None` or `{}` sentinel with no logging. The result is that exported files contain `None` values for metric columns where computation failed — and a consumer of those files has no way to determine whether `None` means "no path exists between these nodes" (a valid data point for a disconnected graph) or "the computation raised an exception and was silently discarded."

For a project explicitly positioned as infrastructure for regulated, auditable AI systems, data correctness in exported files is a first-order concern. A downstream ML pipeline, compliance report, or analyst treating every `None` as "no relationship" would systematically misuse failed-computation rows as valid training or audit data.

## Reproduction

**Four affected methods, exact locations:**

```python
# semantica/export/distance_exporter.py

# Line 72–74: _betweenness()
try:
    result = self._centrality.calculate_betweenness_centrality(graph_dict)
    return result.get("betweenness", {}) if isinstance(result, dict) else {}
except Exception:
    return {}          # ← silent, no log

# Line 82–84: _hop_distance()
try:
    result = self._path_finder.bfs_shortest_path(graph_dict, src, tgt)
    path = result.get("path", []) if isinstance(result, dict) else (result or [])
    return len(path) - 1 if path else None
except Exception:
    return None        # ← silent, no log

# Line 93–95: _weighted_distance()
try:
    result = self._path_finder.dijkstra_shortest_path(graph_dict, src, tgt)
    if isinstance(result, dict):
        return float(result.get("total_weight", len(result.get("path", [])) - 1))
    return None
except Exception:
    return None        # ← silent, no log

# Line 102–104: _semantic_similarity()
try:
    sim = self._similarity.cosine_similarity(graph_dict, src, tgt)
    return float(sim) if isinstance(sim, (int, float)) else None
except Exception:
    return None        # ← silent, no log
```

**Call chain into `compute_pairs()`** (lines 118–167):

```python
def compute_pairs(self, include=None, node_subset=None):
    ...
    if "hop_count" in include_set or "distance_band" in include_set:
        hop_count = self._hop_distance(graph_dict, src, tgt)  # silently None on error
        if "hop_count" in include_set:
            row["hop_count"] = hop_count    # written to row as-is

    if "weighted_distance" in include_set:
        row["weighted_distance"] = self._weighted_distance(...)  # silently None on error

    if "semantic_similarity" in include_set:
        row["semantic_similarity"] = self._semantic_similarity(...)  # silently None on error
```

**The ambiguity — `hop_count` specifically:**

`_hop_distance()` returns `None` in two distinct cases:
1. `bfs_shortest_path()` returns a path of length 0 or no path — `len([]) - 1` is never reached; the `if path else None` branch fires. This is **valid data**: no path between the nodes.
2. `bfs_shortest_path()` raises an exception — caught, `return None`.

A consumer reading `hop_count=None` in the exported CSV cannot distinguish these.

## Root Cause

All four helper methods follow the same pattern: bare `except Exception: return <sentinel>` with no `logger.warning()` or `logger.debug()` call, no exception message preserved, and no mechanism for `compute_pairs()` to learn that a computation failed. The exported row is written identically whether the metric computed correctly (and was `None` by value) or silently failed.

## Suggested Fix

Two complementary changes:

**1. Log the exception in each helper before returning the sentinel:**

```python
def _hop_distance(self, graph_dict, src, tgt):
    try:
        ...
    except Exception as exc:
        logger.debug(
            "hop_distance(%s, %s) failed: %s — returning None (caller cannot "
            "distinguish this from 'no path exists')", src, tgt, exc
        )
        return None
```

This is the minimum change: operators with debug logging enabled will see failed computations in the log, even if the exported data still shows `None`.

**2. Consider an explicit error/status column per row:**

For use cases where exported data feeds downstream systems (ML training sets, compliance reports, data lineage audits), a per-metric error flag would let consumers filter or flag failed rows:

```python
row["hop_count_error"] = None     # set to str(exc) if computation failed
row["semantic_similarity_error"] = None
```

Or a single `metric_errors` column listing which metrics failed for a given pair. This is an API/format change that may need a discussion on the preferred schema, but it closes the ambiguity for downstream consumers in a way that logging alone does not.

### PR DESCRIPTION
**Pull request by AmirF194**
## What's wrong

`DistanceExporter`'s four private metric helpers (`_betweenness`, `_hop_distance`, `_weighted_distance`, `_semantic_similarity`) each wrap their computation in a bare `except Exception: return None`/`{}` with no logging. An exported `None` is therefore indistinguishable from a legitimate "no path exists between these nodes" result, so a consumer of the exported CSV/JSONL/DataFrame data can't tell "computation raised and was silently discarded" from "there really is no path here."

## Fix

Log each caught exception at warning level (with the source/target and full traceback) before returning the existing sentinel. The exported row shape and values are unchanged; only the observability of the failure changes. This is the minimal fix from the two options discussed in the issue (log-only vs. a schema-changing per-metric error column); the schema change is a separate, larger design decision better left for its own discussion.

## Verification

- New regression test (`tests/export/test_distance_exporter.py`) reproduces the bug: with the four helpers monkeypatched to raise, the branch logs a warning per failure while `main` logs nothing. Confirmed both directions (fails on `main`, passes on this branch).
- Also asserts the exported row shape/values are unchanged, and that the pre-existing "no KG backend available" path (a legitimate `None`, not an exception) still logs nothing.
- Ran the full `tests/export/` suite plus the other export-adjacent suites (`test_export_module.py`, `test_arango_aql_exporter.py`, `test_401_temporal_provenance_export.py`, `test_parquet_exporter.py`, `test_export_methods_wrapper.py`, `test_arrow_exporter.py`, `test_notebook_15_export.py`): all pass.
- Did not run the full repository test suite; the rest of it requires heavier optional dependencies (torch/transformers/sklearn/etc.) not installed in this verification environment.

Fixes #874

### CODE REVIEW

**Comment by KaifAhmad1** [comment]
@AmirF194 Please fix the Qodo findings first, especially the logger mismatch and misleading warning messages. Once addressed, we can review and proceed with the merge.

**Comment by AmirF194** [comment]
Pushed a fix for both. `get_logger(__name__)` was doubling the `semantica.` prefix (`__name__` is already `semantica.export.distance_exporter`), so the warnings landed on `semantica.semantica.export.distance_exporter`, a branch `setup_logging()` never configures. Changed it to `get_logger("export.distance_exporter")`, confirmed in a clean container that `logger.name` now resolves to `semantica.export.distance_exporter`. Also rewrote the three exception-path messages (hop/weighted/semantic) to say "returning None sentinel" instead of "recording as no path", since a raised exception isn't the same claim as a missing path.

Left the other two Qodo findings alone on purpose: adding node context to the betweenness warning doesn't fit cleanly, that computation runs over the whole graph rather than a single src/tgt pair, so there's no pair to name. And the per-metric error/status export field is the schema-changing alternative Qodo itself flagged as a separate design call, not the log-only fix this PR and the issue scoped. Happy to open a follow-up for that if you want it.

Full `tests/export/` plus the other export suites still green (175 passed) after the change.

**Comment by Sameer6305** [comment]
Hi @AmirF194 , if you're working on an issue, please leave a comment on the issue mentioning that you're handling it. This will help us keep track of engaged issues and assign them accordingly. Thanks!

**Comment by AmirF194** [comment]
#890 opened a few hours after this one, same issue, same four call sites. One difference worth flagging: it logs at debug level, this PR logs at warning with exc_info, which matches what #874 actually asks for (the exception was invisible by default before either fix; debug-level stays invisible at any commonly deployed log level). Happy to stand down if you'd rather go with #890.

**Review by KaifAhmad1 — APPROVED** [review]
Thanks for the thorough fix and follow-through here, @AmirF194 — the logger double-prefix catch in particular was a good find, and the reasoning for leaving the betweenness context/error-column suggestions for a follow-up was sound.

Verified independently:
- `get_logger("export.distance_exporter")` now resolves to `semantica.export.distance_exporter`, matching every other exporter and fixing a pre-existing silent `logger.debug` call as a bonus.
- `tests/export/test_distance_exporter.py` (6 tests) and the full `tests/export/` suite (71 tests) pass on this branch.
- Exported row shape/sentinel values are unchanged; only observability improved, as intended.

Added a changelog entry crediting you in a follow-up commit. Approving.
